hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -176,7 +177,10 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute(HoodieData<HoodieRec
     context.setJobStatus(this.getClass().getSimpleName(), "Doing partition and writing data: " + config.getTableName());
     HoodieData<WriteStatus> writeStatuses = mapPartitionsAsRDD(inputRecordsWithClusteringUpdate, partitioner);
     HoodieWriteMetadata<HoodieData<WriteStatus>> result = new HoodieWriteMetadata<>();
-    updateIndexAndCommitIfNeeded(writeStatuses, result);
+    // dereference rdd so that no double de-referencing can happen by mistake.
+    int numPartitions = Math.max(1, writeStatuses.getNumPartitions());
+    HoodieData<WriteStatus> computedWriteStatus = HoodieJavaRDD.of(writeStatuses.collectAsList(), (HoodieSparkEngineContext) context, numPartitions);
```

Comment:
If write status tracking success record, will collecting them in driver case pressure on driver memory?

Reply:
+1 This may not work in case of large inserts with millions of inserts. Success record tracking will be enabled when certain indexes (e.g. record index) is enabled in the MDT.